x/profiles/keeper/keeper_relationships.go

```diff
@@ -1,37 +1,46 @@
 package keeper
 
 import (
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 
 	"github.com/desmos-labs/desmos/x/profiles/types"
 )
 
 // SaveRelationship allows to store the given relationship returning an error if he's already present.
 func (k Keeper) SaveRelationship(ctx sdk.Context, relationship types.Relationship) error {
 	store := ctx.KVStore(k.storeKey)
-	key := types.RelationshipsStoreKey(relationship.Creator)
+	key := types.RelationshipsStoreKey(relationship.Creator, relationship.Subspace, relationship.Recipient)
 
-	relationships := types.MustUnmarshalRelationships(k.cdc, store.Get(key))
-	for _, rel := range relationships {
-		if rel.Equal(relationship) {
-			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest,
-				"relationship already exists with %s", relationship.Recipient)
-		}
+	if store.Has(key) {
+		return fmt.Errorf("the relationship has already existed")
 	}
 
-	relationships = append(relationships, relationship)
-	store.Set(key, types.MustMarshalRelationships(k.cdc, relationships))
+	store.Set(key, types.MustMarshalRelationship(k.cdc, relationship))
 	return nil
 }
 
+func (k Keeper) GetRelationship(ctx sdk.Context, creator, subspace, recipient string) (types.Relationship, bool) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.RelationshipsStoreKey(creator, subspace, recipient)
+
+	if !store.Has(key) {
+		return types.Relationship{}, false
+	}
+
+	bz := store.Get(key)
+
+	var rel types.Relationship
+	k.cdc.MustUnmarshalBinaryBare(bz, &rel)
+	return rel, true
```

Comment:
We can simplify this by not declaring `bz` and using the `types.MustUnmarshalRelationship` method: 
```suggestion
	return types.MustUnmarshalRelationship(k.cdc, store.Get(key)), true
```